allennlp_rc/eval/orb_eval.py

```diff
@@ -0,0 +1,88 @@
+"""Official evaluation script for ORB.
+Usage:
+  python evaluation_script.py
+        --dataset_file <file_path>
+        --prediction_file <file_path>
+        --metrics_output_file <file_path>
+"""
+
+# pylint: disable=unused-import
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import json
+from allennlp_rc.eval.orb_utils import evaluate_dataset
+from allennlp_rc.eval.narrativeqa_eval import evaluate_narrativeqa
+
+
+def read_predictions(json_file):
+    return json.load(open(json_file))
+
+
+def read_labels(jsonl_file):
+    qid_answer_map = {}
+    with open(jsonl_file) as f:
+        for line in f:
+            data = json.loads(line)
+            for qa_pair in data["qa_pairs"]:
+                qid_answer_map[str(qa_pair["qid"])] = {"dataset": qa_pair["dataset"],
+                                                       "answers": qa_pair["answers"]}
+    return qid_answer_map
+
+
+def compute_averages(all_metrics):
+    for dataset, dataset_metric in all_metrics.items():
+        if len(dataset_metric) > 0:
+            total = dataset_metric["total"]
+            for metric, value in dataset_metric.items():
+                if metric != "total":
+                    dataset_metric[metric] = value / float(total)
+    return all_metrics
+
+
+def evaluate(answers, predictions):
+    metrics = {"drop": {}, "squad1": {}, "squad2": {}, "newsqa": {},
+               "quoref": {}, "ropes": {}, "narrativeqa": {}, "duorc": {}}
+    for qid, ground_truth_dict in answers.items():
+        if qid in predictions:
+            predicted_answer = predictions[qid]
+            dataset_name = ground_truth_dict["dataset"].lower()
+            try:
+                if dataset_name == "narrativeqa":
+                    metrics = evaluate_narrativeqa(predicted_answer, ground_truth_dict["answers"], metrics)
```

Comment:
This should just be an `if` inside of `evaluate_dataset`, like all of the others.